libvips/iofuncs/init.c

```diff
@@ -720,6 +720,9 @@ vips_shutdown( void )
 	gsf_shutdown(); 
 #endif /*HAVE_GSF*/
 
+       VIPS_FREE(vips__argv0);
+       g_set_prgname(NULL);
```

Comment:
libvips probably shouldn't call g_set_prgname(), or unset it either. I think it's supposed to be controlled at the application level, eg. by GApplication.

I'm happy to merge your other suggestions.

Reply:
I made a commit following your comment so now you can merge it.

I had trouble amending my commit and ended up with two commits, sorry about that.
If you can "Squash and merge" my commits, that'd be great.